src/Domain/Creators/BinaryCompactObject.hpp

```diff
@@ -264,7 +266,23 @@ class BinaryCompactObject : public DomainCreator<3> {
         "The time dependence of the moving mesh domain."};
   };
 
-  using options = tmpl::list<
+  template <typename BoundaryConditionsBase>
+  struct InnerBoundaryCondition {
+    static std::string name() noexcept { return "InnerBoundaryCondition"; }
+    static constexpr Options::String help =
+        "Options for the inner boundary conditions.";
+    using type = std::unique_ptr<BoundaryConditionsBase>;
+  };
+
+  template <typename BoundaryConditionsBase>
+  struct OuterBoundaryCondition {
+    static std::string name() noexcept { return "OuterBoundaryCondition"; }
```

Comment:
I think you meant to name this "InnerBoundary", and the one below "OuterBoundary", and put them in a `BoundaryConditionsGroup`, like you did for `Interval` (I like the grouping you did there).